src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -861,7 +861,13 @@ ExecuteUtilityTaskListWithoutResults(List *taskList, bool localExecutionSupporte
 	/* execute remote tasks if any */
 	if (list_length(remoteTaskList) > 0)
 	{
-		ExecuteTaskList(rowModifyLevel, remoteTaskList, MaxAdaptiveExecutorPoolSize);
+		/*
+		 * We already executed tasks locally. We should ideally remove this method and
+		 * let ExecuteTaskListExtended handle the local execution.
```

Comment:
This would be good to address.

Reply:
I was thinking of another PR for that as that might be some work, what do you think?

Reply:
yes, that's fine.